colour/io/luts/sony_spimtx.py

```diff
@@ -13,8 +13,7 @@
 import typing
 
 if typing.TYPE_CHECKING:
-    from pathlib import Path
-    from io import StringIO
+    from _typeshed import StrPath
```

Comment:
Sorry @jorenham for the late reply, it seems like my comment did not go through for some reason, I might have closed the laptop lid before they got pushed. At any rate, what I wrote was along the lines of I don't think we should rely on a type from a private import. In general, and unless there is a specific reason, we try to avoid doing this. I would prefer that we revert back to `str | Path` here.

Reply:
`_typeshed` is bit of a special case that's maintained by Python itself:  https://github.com/python/typeshed/tree/main/stdlib/_typeshed

And although they don't officially guarantee permanent api stability, `StrPath` is one that's used *a lot*. For example, the Python standard library itself uses it 56 times: https://github.com/search?q=repo%3Apython%2Ftypeshed+StrPath+path%3A%2F%5Estdlib%5C%2F%2F&type=code, and libraries like NumPy also use it: https://github.com/search?q=repo%3Anumpy%2Fnumpy+StrPath&type=code

I know that the typeshed maintainers are also very careful about these sort of things. Any change they make will be ran through [mypy_primer](https://github.com/hauntsaninja/mypy_primer), and `colour` is listed in its projects. This means that if a typeshed change will result in a new typing error in `colour`, then it's likely that they will not go through with it.

Reply:
Ok, this is reasonable then. I suppose it is only available when type checking? If so, two paths:

1. Either we keep using `str | Path`
2. We alias it in `colour.hints.__init__` and use it across the board. We have `str | Path` / `Path | str` used in a few places.

Reply:
```suggestion
    from _typeshed import StrPath
```

Yea; `_typeshed` is a type-checking only module, and type-checkers consider it as if it's part of the standard library. But since you're using `__future__.annotations`, that's not much of a problem. 

> 2. We alias it in `colour.hints.__init__` and use it across the board. We have `str | Path` / `Path | str` used in a few places.

I suppose that you could re-export it if you want to, but theoretically that's not much different from importing it directly 🤷🏻.

But either way, this is all pretty much out-of-scope for this PR, so option 1 it is